oauth2/src/main/scala/tsec.oauth2/provider/DataHandler.scala

```diff
@@ -0,0 +1,40 @@
+package tsec.oauth2.provider
+
+import java.util.Date
+
+import scala.concurrent.duration._
+import scala.concurrent.duration.FiniteDuration
+
+/**
+ * Provide accessing to data storage for using OAuth 2.0.
+ */
+trait DataHandler[U] extends AuthorizationHandler[U] with ProtectedResourceHandler[U]
+
+/**
+ * Access token
+ *
+ * @param token Access token is used to authentication.
+ * @param refreshToken Refresh token is used to re-issue access token.
+ * @param scope Inform the client of the scope of the access token issued.
+ * @param life Life of the access token since its creation.
+ * @param createdAt Access token is created date.
+ * @param params Additional parameters to add information/restriction on given Access token.
+ */
+final case class AccessToken(token: String, refreshToken: Option[String], scope: Option[String], life: Option[FiniteDuration], createdAt: Date, params: Map[String, String] = Map.empty[String, String]) {
```

Comment:
why `Date` and not `Instant`? The latter is what people use nowadays. Not like we're worried about supporting jdk7 and below: we don't even build for it.

Reply:
Also I'd prefer `lifeTime` over `life` here.